### PR TITLE
Add DiagnosticTag support

### PR DESCRIPTION
**Pull request by akaroml**
Signed-off-by: Rome Li <rome.li@microsoft.com>

### CODE REVIEW

**Comment by akaroml** [comment]
ref: https://github.com/microsoft/vscode-languageserver-node/blob/082631da44d6799358602a4abebadbf4fc47859b/types/src/main.ts#L513

File: org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/DiagnosticTag.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ ******************************************************************************/
+
+package org.eclipse.lsp4j;
+
+public enum DiagnosticTag {
+	
+	/**
+	 * Unused or unnecessary code.
+     *
```
Comment (yaohaizh) [inline org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/DiagnosticTag.java]: skewed comments

File: org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/DiagnosticTag.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	 * Unused or unnecessary code.
+     * 
+     * Clients are allowed to render diagnostics with this tag faded out instead of having
+	 * an error squiggle.
+	 */
+    Unnecessary(1),
+    
+	/**
+	 * Deprecated or obsolete code.
+     *
```
Comment (yaohaizh) [inline org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/DiagnosticTag.java]: Same here

File: org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/DiagnosticTag.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	/**
+	 * Deprecated or obsolete code.
+     * 
+     * Clients are allowed to rendered diagnostics with this tag strike through.
+	 */
+    Deprecated(2);
+
+	private final int value;
+	
+	DiagnosticTag(int value) {
```
Comment (yaohaizh) [inline org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/DiagnosticTag.java]: Access modifier for constuctor

Comment (akaroml) [reply org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/DiagnosticTag.java]: Just checked other places. No access modifiers were put on enum types.

**Comment by akaroml** [comment]
Fixed indentations

File: org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/DiagnosticTag.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
+/******************************************************************************
+ * Copyright (c) 2019 Mcirosoft.
```
Comment (spoenemann) [inline org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/DiagnosticTag.java]: I guess the copyright owner should be spelled differently?

**Comment by KamasamaK** [comment]
@spoenemann Yes, in v3.15. See microsoft/language-server-protocol#500

File: org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/DiagnosticTag.java (hunk 4 above)
Comment (akaroml) [reply org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/DiagnosticTag.java]: You've got me! 👍 

Just fixed.

**Comment by spoenemann** [comment]
Thanks!

**Pull request merged**